### PR TITLE
selfcheck: fix messages translations

### PR DESCRIPTION
**Pull request by lauren-d**
selfcheck translations needs to work outside of the application context.

* Uses `force_locale` method from babel package.
* Closes: rero#3423.

### CODE REVIEW

File: rero_ils/modules/selfcheck/api.py [hunk 1]
```diff
@@ -192,71 +198,78 @@ def patron_information(barcode, **kwargs):
...
-                home_phone=patron_dumps.get('home_phone'),
-                home_address=format_patron_address(patron),
-                currency_type=patron.organisation.get('default_currency'),
-                valid_patron=patron.is_patron
-            )
+        language = kwargs.get('language', current_app.config
+                              .get('BABEL_DEFAULT_LANGUAGE'))
+        # Temporarily overrides the currently selected locale.
+        # force_locale is allowed to work outside of application context
+        with ((((force_locale(language))))):
```
Comment (rerowep) [inline rero_ils/modules/selfcheck/api.py:205]: Why so many parenthesis?
```python
with force_locale(language):
```

File: rero_ils/modules/selfcheck/api.py [hunk 2]
```diff
@@ -273,10 +286,11 @@ def item_information(item_barcode, **kwargs):
...
         item = Item.get_item_by_barcode(item_barcode, org_pid)
-        with current_app.test_request_context() as ctx:
-            language = kwargs.get('language', current_app.config
-                                  .get('BABEL_DEFAULT_LANGUAGE'))
-            ctx.babel_locale = language
+        language = kwargs.get('language', current_app.config
+                              .get('BABEL_DEFAULT_LANGUAGE'))
+        # Temporarily overrides the currently selected locale.
+        # force_locale is allowed to work outside of application context
+        with force_locale(language):
```
Comment (rerowep) [inline rero_ils/modules/selfcheck/api.py:293]: Maybe better to change to:
```python
if item:
    with force_locale(language):
        document = Document.get_record_by_pid(item.document_pid)
        location = item.get_location()
```

File: rero_ils/modules/selfcheck/api.py (hunk 1 above)
Comment (Garfield-fr) [inline rero_ils/modules/selfcheck/api.py:205]: Why multiple `(` ?

File: rero_ils/modules/selfcheck/api.py [hunk 3]
```diff
@@ -192,71 +198,78 @@ def patron_information(barcode, **kwargs):
...
+        language = kwargs.get('language', current_app.config
+                              .get('BABEL_DEFAULT_LANGUAGE'))
+        # Temporarily overrides the currently selected locale.
+        # force_locale is allowed to work outside of application context
+        with ((((force_locale(language))))):
+            patron = Patron.get_patron_by_barcode(
+                barcode=barcode, org_pid=institution_id)
+            if patron:
+                from pprint import pprint
+                pprint(patron)
```
Comment (Garfield-fr) [inline rero_ils/modules/selfcheck/api.py:210]: Delete this

**Review by Garfield-fr — APPROVED** [review]

File: rero_ils/modules/selfcheck/api.py [hunk 4]
```diff
@@ -503,7 +520,7 @@ def selfcheck_checkin(transaction_user_pid, item_barcode, **kwargs):
                     _('No circulation action is possible'))
             except CirculationException as circ_err:
                 checkin.get('screen_messages', []).append(
-                    _(circ_err.description))
+                    circ_err.description)
```
Comment (rerowep) [inline rero_ils/modules/selfcheck/api.py:523]: Why no translation for `circ_err.description`? 
(https://github.com/pallets/werkzeug/blob/main/src/werkzeug/exceptions.py)

**Review by PascalRepond — APPROVED** [review]

File: rero_ils/modules/selfcheck/api.py (hunk 2 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py:291]: I need `force_locale` for the else statement

File: rero_ils/modules/selfcheck/api.py (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: Uuups 🫣

File: rero_ils/modules/selfcheck/api.py (hunk 3 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: Done

File: rero_ils/modules/selfcheck/api.py (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: I need glasses... 🤪

**Pull request merged**